src/move/econia/sources/user.move

```diff
@@ -1438,6 +1526,155 @@ module econia::user {
+    }
+
+    public(friend) fun emit_place_limit_order_event(
+        event: PlaceLimitOrderEvent
+    ) acquires MarketEventHandles {
+        let user_address = event.user;
+        if (!exists<MarketEventHandles>(user_address)) return;
+        let market_event_handles_ref_mut =
+            borrow_global_mut<MarketEventHandles>(user_address);
+        let market_account_id =
+            get_market_account_id(event.market_id, event.custodian_id);
+        let place_limit_order_events_ref_mut =
+            &mut market_event_handles_ref_mut.place_limit_order_events;
+        let has_handle = tablist::contains(
+            place_limit_order_events_ref_mut, market_account_id);
+        if (!has_handle) return;
+        let place_limit_order_event_handle_ref_mut = tablist::borrow_mut(
+            place_limit_order_events_ref_mut, market_account_id);
+        event::emit_event(
+            place_limit_order_event_handle_ref_mut, event)
+    }
+
+    public(friend) fun emit_place_market_order_event(
+        event: PlaceMarketOrderEvent
+    ) acquires MarketEventHandles {
+        let user_address = event.user;
+        if (!exists<MarketEventHandles>(user_address)) return;
+        let market_event_handles_ref_mut =
+            borrow_global_mut<MarketEventHandles>(user_address);
+        let market_account_id =
+            get_market_account_id(event.market_id, event.custodian_id);
+        let place_market_order_events_ref_mut =
+            &mut market_event_handles_ref_mut.place_market_order_events;
+        let has_handle = tablist::contains(
+            place_market_order_events_ref_mut, market_account_id);
+        if (!has_handle) return;
+        let place_market_order_event_handle_ref_mut = tablist::borrow_mut(
+            place_market_order_events_ref_mut, market_account_id);
+        event::emit_event(
+            place_market_order_event_handle_ref_mut, event)
+    }
+
+    public(friend) fun pack_fill_event(
+        market_id: u64,
+        size: u64,
+        price: u64,
+        maker_side: bool,
+        maker: address,
+        maker_custodian_id: u64,
+        maker_order_id: u128,
+        taker: address,
+        taker_custodian_id: u64,
+        taker_order_id: u128,
+    ): FillEvent {
+        FillEvent{
+            market_id,
+            size,
+            price,
+            maker_side,
+            maker,
+            maker_custodian_id,
+            maker_order_id,
+            taker,
+            taker_custodian_id,
+            taker_order_id
+        }
+    }
+
+    public(friend) fun get_fill_event_taker_order_id(
+        fill_event_ref: &FillEvent,
+    ): u128 {
+        fill_event_ref.taker_order_id
+    }
+
+    public(friend) fun set_fill_event_taker_order_id(
+        fill_event_ref_mut: &mut FillEvent,
+        taker_order_id: u128
+    ) {
+        fill_event_ref_mut.taker_order_id = taker_order_id
+    }
+
+    public(friend) fun pack_place_limit_order_event(
```

Comment:
Why pack?

Reply:
https://move-language.github.io/move/structs-and-resources.html?highlight=pack#creating-structs

Reply:
No one says that. Why not create?